packages/ui/src/machines/authMachine.ts

```diff
@@ -110,9 +112,33 @@ export const authMachine = createMachine<AuthContext, AuthEvent>(
       clearUser: assign({
         user: undefined,
       }),
-      setAuthConfig: assign({
-        config(_, event) {
-          return event.data.auth;
+      applyAmplifyConfig: assign({
+        config(context, event) {
+          const loginMechanisms = event.data.aws_cognito_login_mechanisms?.map(
+            (login) => {
+              switch (login) {
+                case 'PREFERRED_USERNAME':
+                case 'USERNAME':
+                  return 'username';
```

Comment:
@lazpavel So this should just be:

```suggestion
                case 'PREFERRED_USERNAME':
                  return 'username';
```

?

Reply:
yes